xilem/examples/widgets.rs

```diff
@@ -7,50 +7,95 @@ use masonry::dpi::LogicalSize;
 use masonry::event_loop_runner::{EventLoop, EventLoopBuilder};
 use winit::error::EventLoopError;
 use winit::window::Window;
-use xilem::view::{button, checkbox, flex, label, progress_bar, FlexSpacer};
-use xilem::{WidgetView, Xilem};
+use xilem::core::adapt;
+use xilem::view::{button, checkbox, flex, flex_item, progress_bar, sized_box, Axis, FlexSpacer};
+use xilem::{Color, WidgetView, Xilem};
+
+const SPACER_WIDTH: f64 = 10.;
 
 /// The state of the entire application.
 ///
 /// This is owned by Xilem, used to construct the view tree, and updated by event handlers.
 struct WidgetGallery {
     progress: Option<f64>,
+    checked: bool,
 }
 
-fn app_logic(data: &mut WidgetGallery) -> impl WidgetView<WidgetGallery> {
+fn progress_bar_view(data: Option<f64>) -> impl WidgetView<Option<f64>> {
     flex((
-        label("this 'widgets' example currently only has 1 widget"),
-        FlexSpacer::Flex(1.),
-        progress_bar(data.progress),
+        progress_bar(data),
         checkbox(
             "set indeterminate progress",
-            data.progress.is_none(),
-            |state: &mut WidgetGallery, checked| {
+            data.is_none(),
+            |state: &mut Option<f64>, checked| {
                 if checked {
-                    state.progress = None;
+                    *state = None;
                 } else {
-                    state.progress = Some(0.5);
+                    *state = Some(0.5);
                 }
             },
         ),
-        button("change progress", |state: &mut WidgetGallery| {
-            match state.progress {
-                Some(ref mut v) => *v = (*v + 0.1).rem_euclid(1.),
-                None => state.progress = Some(0.5),
-            }
+        button("change progress", |state: &mut Option<f64>| match state {
+            Some(ref mut v) => *v = (*v + 0.1).rem_euclid(1.),
+            None => *state = Some(0.5),
         }),
-        FlexSpacer::Flex(1.),
     ))
 }
 
+fn checkbox_view(data: bool) -> impl WidgetView<bool> {
```

Comment:
```suggestion
fn checkbox_view(checked: bool) -> impl WidgetView<bool> {
```
I'm not convinced that this is idiomatic, however. I'd probably just inline this here.